eval/eval.go

```diff
@@ -258,7 +265,7 @@ func (s *State) evalInternal(node any) object.Object { //nolint:funlen,gocognit,
 			log.LogVf("eval infix %s", node.DebugString())
 		}
 		// Eval and not evalInternal because we need to unwrap "return".
-		if node.Token.Type() == token.ASSIGN || node.Token.Type() == token.DEFINE {
+		if node.Token.Type() == token.ASSIGN || (node.Token.Type() >= token.DEFINE && node.Token.Type() < token.FUNC) {
```

Comment:
same comment and because it's there 2x let's make that change check a function

isAssignement() which is `tokt ==token.ASSIGN  || tokt == token.DEFINE || isCompound(tokt)`
so you can reuse isCompound() above